main/webapp/modules/core/scripts/views/data-table/data-table-view.js

```diff
@@ -92,57 +92,127 @@ DataTableView.resizingState = {
   releaseListener: null, // the event listener for mouse release events
 };
 
+DataTableView.resizingState = {
+  dragging: false,
+  col: null,
+  columnName: null,
+  originalWidth: 0,
+  originalPosition: 0,
+  moveListener: null,
+  releaseListener: null,
+  tableContainer: null, // Add reference to table container
+  columnIndex:0 // Reference to column number
+};
+
 DataTableView.prototype._startResizing = function(columnIndex, clickEvent) {
   var self = this;
   var columnHeader = self._columnHeaderUIs[columnIndex];
-  clickEvent.preventDefault();
   var state = DataTableView.resizingState;
+
+  clickEvent.preventDefault();
   state.dragging = true;
   state.col = columnHeader._col;
   state.columnName = columnHeader._column.name;
   state.originalWidth = columnHeader._col.width();
   state.originalPosition = clickEvent.pageX;
-  // for conversion from px to em
-  state.emFactor = parseFloat(getComputedStyle($(".data-table-container colgroup")[0]).fontSize);
+  // Store table container reference when starting resize
+  state.tableContainer = this._div.find(".data-table");
+  state.tableWidth = this._div.find(".column-header");
+  state.columnIndex=columnIndex+1;
+
+  // Find the corresponding <th> for this column
+  var thElement = $('.data-table-header th').eq(columnIndex+1);
+  if (thElement.length === 0) {
+    console.error("Column <th> not found for index:", columnIndex);
+    return;
+  }
+
+  var thRect = thElement[0].getBoundingClientRect();
+
+  var themeColor = columnHeader._col.css('border-color');
+
+  $('#column-resize-highlight').remove();
+
+  var highlightLine = $('<div></div>')
```

Comment:
By proposing a CSS-based solution, I was expecting that there would not be any need for creating a dedicated element to represent the column border. See the details of my proposal in my earlier reply.